crates/miden-tx/tests/integration/scripts/compatibility.rs

```diff
@@ -0,0 +1,490 @@
+        ACCOUNT_ID_REGULAR_ACCOUNT_IMMUTABLE_CODE_ON_CHAIN_2,
+    },
+    Felt,
+};
+use miden_tx::testing::{Auth, MockChain};
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaCha20Rng;
+
+// DEFAULT NOTES
+// ================================================================================================
+
+#[test]
+fn test_basic_wallet_default_notes() {
+    // STATELESS CHECK
+    // --------------------------------------------------------------------------------------------
+
+    let mut mock_chain = MockChain::new();
+    let wallet_account =
+        mock_chain.add_existing_wallet(Auth::BasicAuth, vec![FungibleAsset::mock(20)]);
+    let wallet_account_interface = AccountInterface::from(&wallet_account);
+
+    let faucet_account = mock_chain.add_existing_faucet(Auth::BasicAuth, "POL", 200u64, None);
+    let faucet_account_interface = AccountInterface::from(faucet_account.account());
+
+    let p2id_note = create_p2id_note(
+        ACCOUNT_ID_REGULAR_ACCOUNT_IMMUTABLE_CODE_ON_CHAIN.try_into().unwrap(),
+        ACCOUNT_ID_REGULAR_ACCOUNT_IMMUTABLE_CODE_ON_CHAIN_2.try_into().unwrap(),
+        vec![FungibleAsset::mock(10)],
+        NoteType::Public,
+        Default::default(),
+        &mut RpoRandomCoin::new([Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)]),
+    )
+    .unwrap();
+
+    let p2idr_note = create_p2idr_note(
+        ACCOUNT_ID_REGULAR_ACCOUNT_IMMUTABLE_CODE_ON_CHAIN.try_into().unwrap(),
+        ACCOUNT_ID_REGULAR_ACCOUNT_IMMUTABLE_CODE_ON_CHAIN_2.try_into().unwrap(),
+        vec![FungibleAsset::mock(10)],
+        NoteType::Public,
+        Default::default(),
+        BlockNumber::default(),
+        &mut RpoRandomCoin::new([Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)]),
+    )
+    .unwrap();
+
+    let offered_asset = NonFungibleAsset::mock(&[5, 6, 7, 8]);
+    let requested_asset = NonFungibleAsset::mock(&[1, 2, 3, 4]);
+
+    let (swap_note, _) = create_swap_note(
+        ACCOUNT_ID_REGULAR_ACCOUNT_IMMUTABLE_CODE_ON_CHAIN.try_into().unwrap(),
+        offered_asset,
+        requested_asset,
+        NoteType::Public,
+        Felt::new(0),
+        &mut RpoRandomCoin::new([Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)]),
+    )
+    .unwrap();
+
+    // Basic wallet
+    assert_eq!(CheckResult::Yes, wallet_account_interface.can_consume(&p2id_note));
+    assert_eq!(CheckResult::Yes, wallet_account_interface.can_consume(&p2idr_note));
+    assert_eq!(CheckResult::Yes, wallet_account_interface.can_consume(&swap_note));
+
+    // Basic fungible faucet
+    assert_eq!(CheckResult::No, faucet_account_interface.can_consume(&p2id_note));
+    assert_eq!(CheckResult::No, faucet_account_interface.can_consume(&p2idr_note));
+    assert_eq!(CheckResult::No, faucet_account_interface.can_consume(&swap_note));
+
+    // STATEFUL CHECK
+    // --------------------------------------------------------------------------------------------
+
+    // TODO: implement
```

Comment:
I think I would leave the tests as-is and add separate "stateful check tests", so that the tests don't become very large. The smaller a test is, the easier it is to debug it if it fails.

Once we add stateful checks, we can move the test setup code into a shared setup function to avoid code duplication.